### ISSUE TITLE
Saving a project with zero changes causes two design-time builds

### ISSUE THREAD

**Issue by davkean** [dotnet/project-system#2166]
1. Right-click on a project -> Edit 
2. CTRL+S

Expected: A single design-time build
Actual:

```
15>------ Build started: Project: ConsoleApp303, Targets: CollectResolvedSDKReferencesDesignTime, CollectPackageReferences, ResolveComReferencesDesignTime, ResolveProjectReferencesDesignTime, ResolveAssemblyReferencesDesignTime, CollectAnalyzersDesignTime, CollectSDKReferencesDesignTime, ResolvePackageDependenciesDesignTime, CompileDesignTime ------
15>------ Build completed ------
16>------ Build started: Project: ConsoleApp303, Targets: CollectResolvedSDKReferencesDesignTime, CollectPackageReferences, ResolveComReferencesDesignTime, ResolveProjectReferencesDesignTime, ResolveAssemblyReferencesDesignTime, CollectAnalyzersDesignTime, CollectSDKReferencesDesignTime, ResolvePackageDependenciesDesignTime, CompileDesignTime ------
16>------ Build completed ------
```

**Comment by davkean** [comment]
It appears that first build occurs on an older version of the project before the Save.

**Comment by davkean** [comment]
@lifengl What's going on here? I'm seeing this with any change I make to the project, we're kicking off multiple design-time builds one with older data and one with updated data.

**Comment by lifengl** [comment]
I saw that too.  I made a change recently to reduce it, but still builds twice.  The build is driven by data flow, and triggered twice through different routes.  Maybe one comes from a capability provider.

We have a existing bug tracking this.

Sent from my phone

On May 18, 2017, at 10:34 PM, David Kean <notifications@github.com<mailto:notifications@github.com>> wrote:


@lifengl<https://github.com/lifengl> What's going on here? I'm seeing this with any change I make to the project, we're kicking off multiple design-time builds one with older data and one with updated data.
